activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -642,6 +644,14 @@ def first_nth_or_last(type, *args)
             set_inverse_instance record if record.is_a? ActiveRecord::Base
           end
         end
+
+        def associate_all_records(target)
+          if owner.new_record? && target.all? {|record| record.new_record? }
```

Comment:
> Car.new(seats: [Seat.new, Seat.find(1)]) <- what do you expect from this scenario?

I'd expect none of the seats to be associated to the car, and I believe the above implementation would not associate any of those seats to the car. 

> but IMO whatever behavior we support should be well-defined and tested.

Yes the new behaviour that I've introduced should be well defined and tested.